### PR TITLE
renovatebot: add automerge presets

### PR DESCRIPTION
**Pull request by kang-makes**
These are the first set of presets that might cover almost all our use cases.

### CODE REVIEW

**Review by roobre — APPROVED** [review]
Looking good overall, I might suggest shortening the names a bit:
- `automerge-all`
- `automerge-minor`
- `automerge-patch`
- `automerge-docker-all`
- `automerge-docker-digest`
- `automerge-docker-minor`
- `automerge-docker-patch`
- `automerge-workflows-all`

Also, newlines 😈

File: renovate/automerge-all-package-all-versions-bump.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        "**/tests/**",
+        "**/__fixtures__/**"
+    ],
+    "printConfig": true,
+    "packageRules": [
+        {
+            "automerge": true,
+            "automergeType": "branch",
+            "pruneBranchAfterAutomerge": true
+        }
```
Comment (roobre) [inline renovate/automerge-all-package-all-versions-bump.json:19]: Keep in mind that this won't work without at least one test passing on the CI.

File: renovate/automerge-all-package-version-minor-bump.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        "**/tests/**",
+        "**/__fixtures__/**"
+    ],
+    "printConfig": true,
+    "packageRules": [
+        {
+            "matchUpdateTypes": [
+                "minor"
+            ],
+            "matchCurrentVersion": "!/^(0|v0)/",
```
Comment (roobre) [inline renovate/automerge-all-package-version-minor-bump.json:19]: Does renovate support negative regexes?

Alternatively:
```suggestion
        {
            "matchUpdateTypes": [
                "minor"
            ],
            "matchCurrentVersion": "/^v?[1-9]/",
```

File: renovate/automerge-github-workflows-all-versions-bump.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "printConfig": true,
+    "packageRules": [
+        {
+            "automerge": true,
+            "automergeType": "branch",
+            "pruneBranchAfterAutomerge": true,
+            "matchDatasources": [
+                "github-tags",
+                "github-runners"
+            ],
```
Comment (roobre) [inline renovate/automerge-github-workflows-all-versions-bump.json:22]: Perhaps matching the github actions manager would be better here
```suggestion
            "matchManagers": ["github-actions"],
```

File: renovate/automerge-all-package-version-minor-bump.json (hunk 2 above)
Comment (kang-makes) [reply renovate/automerge-all-package-version-minor-bump.json:19]: It does. We are using it at New Relic: https://github.com/newrelic/coreint-automation/blob/af3a745b9dffa3c74f916d394e21b463615a5b17/renovate-base.json5#L35

It works :)

**Pull request merged**